components/cam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -699,6 +795,103 @@ subroutine get_cloud_fraction(its,ite,kts,kte,ast,qc,qr,qi,method, &
+       use cam_pio_utils,    only: cam_pio_openfile,cam_pio_closefile
+       use cam_grid_support, only: cam_grid_check, cam_grid_id
+       use cam_grid_support, only: cam_grid_get_dim_names
+       use ncdio_atm,        only: infld
+       use time_manager,     only: get_curr_date
+       use ppgrid,           only: begchunk, endchunk, pcols, pver, pverp, psubcols
+
+      !INOUT/OUTPUT VARIABLES
+      integer,intent(in) :: its,ite,kts,kte,lchnk
+      real(rtype),dimension(its:ite,kts:kte),intent(inout)  :: nccn_prescribed
+      character*(*), intent(in)    :: micro_p3_lookup_dir                !directory of the lookup tables
+      type(physics_buffer_desc),   pointer       :: pbuf(:)
+
+
+      !internal variables
+      character(len=100) :: base_file_name
+      character(len=500) :: filename
+      character(len=20) :: mon_str
+      character(len=20) :: dim1name, dim2name
+      type(file_desc_t) :: nccn_ncid
+      integer :: year, month, day, tod, next_month, grid_id
+      logical :: found = .false.
+      real(rtype),dimension(its:ite,kts:kte) :: nccn_next_month
+      real(rtype) :: fraction_of_month
+      real(rtype), pointer :: mon_ccn_1(:,:)
+      real(rtype), pointer :: mon_ccn_2(:,:)
+      real(rtype), pointer :: ccn_values(:,:,:)
+
+      nullify(ccn_values)
+
+     !get current time step's date
+      call get_curr_date(year,month,day,tod)
+
+     !populate mon_ccn_1 and mon_ccn_2 with corresponding pbuf variables
+
+     call pbuf_get_field(pbuf,mon_ccn_1_idx, mon_ccn_1)
+
+     call pbuf_get_field(pbuf,mon_ccn_2_idx, mon_ccn_2)
+
+     if (current_month .ne. month) then
+
+         mon_ccn_1 = mon_ccn_2
+
+         if (month==12) then
+            next_month = 1
+         else
+            next_month = month + 1
+         end if
+
+         write(mon_str,*) next_month
+
+         mon_str = adjustl(mon_str)
+
+         !assign base_file_name the name of the CCN file being used 
+         base_file_name = "prescribed_CCN_file_"
+
+         !retrieve the name of the relevant file by combining base file name with
+         !month and full file path:
+         filename = trim(micro_p3_lookup_dir)//'/'//trim(base_file_name)//trim(mon_str)//'.nc'
+
+         grid_id = cam_grid_id('physgrid')
+
+         call cam_grid_get_dim_names(grid_id, dim1name, dim2name)
+
+         call cam_pio_openfile(nccn_ncid,filename,PIO_NOWRITE)
+
+         allocate(ccn_values(pcols,pver,begchunk:endchunk))
+
+         call infld('CCN3',nccn_ncid,dim1name,'lev',dim2name,1,pcols,1,pver,begchunk,endchunk,&
+           ccn_values, found, gridname='physgrid')
+
+         call cam_pio_closefile(nccn_ncid)
+
+         mon_ccn_2 = ccn_values(:,:,lchnk)
+
+         deallocate(ccn_values)
+
+         current_month = month
+
+      end if
+
+      !interpolate between mon_ccn_1 and mon_ccn_1 to calculate nccn_prescribed
+      !based on current date
+
+      fraction_of_month = (day*3600.0*24.0 + tod)/(3600*24*30) !tod is in seconds
```

Comment:
I think you're assuming each month has 30 days here - which isn't true? I think it is safe to assume a 365 day calendar with no leap year (at least for SCREAMv0... Luca added leap-year support in the AD)... With a 365 day calendar, you can just create an array of days per month for each month of the year and replace 30 above with days_per_month(month)...

Reply:
Forgot to fix that! Fixed now.